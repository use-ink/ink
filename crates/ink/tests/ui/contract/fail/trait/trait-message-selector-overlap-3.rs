#![allow(unexpected_cfgs)]

mod foo1 {
    #[ink::trait_definition]
    pub trait TraitDefinition1 {
        #[ink(message, selector = 42)]
        fn message1(&self);
    }
}

mod foo2 {
    #[ink::trait_definition]
    pub trait TraitDefinition2 {
        #[ink(message, selector = 42)]
        fn message2(&self);
    }
}

#[ink::contract]
pub mod contract {
    use super::{
        foo1::TraitDefinition1,
        foo2::TraitDefinition2,
    };

    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }
    }

    impl TraitDefinition1 for Contract {
        #[ink(message)]
        fn message1(&self) {}
    }

    impl TraitDefinition2 for Contract {
        #[ink(message)]
        fn message2(&self) {}
    }
}

fn main() {}
