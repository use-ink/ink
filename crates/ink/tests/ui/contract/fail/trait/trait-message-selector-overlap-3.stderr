error[E0119]: conflicting implementations of trait `DispatchableMessageInfo<42>` for type `Contract`
  --> tests/ui/contract/fail/trait/trait-message-selector-overlap-3.rs:43:9
   |
38 |         fn message1(&self) {}
   |         --------------------- first implementation here
...
43 |         fn message2(&self) {}
   |         ^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Contract`

error[E0119]: conflicting implementations of trait `TraitCallForwarderFor<42>` for type `contract::_::CallBuilder`
  --> tests/ui/contract/fail/trait/trait-message-selector-overlap-3.rs:41:5
   |
36 | /     impl TraitDefinition1 for Contract {
37 | |         #[ink(message)]
38 | |         fn message1(&self) {}
39 | |     }
   | |_____- first implementation here
40 |
41 | /     impl TraitDefinition2 for Contract {
42 | |         #[ink(message)]
43 | |         fn message2(&self) {}
44 | |     }
   | |_____^ conflicting implementation for `contract::_::CallBuilder`
